src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -206,6 +209,28 @@
       OutputListLocation="$(OutDir)assemblylist.txt"
       Condition="'$(UseLegacyXunitPerfRunner)'=='true'"
      />
+     
+     <!-- For Desktop Tests, we need to generate Binding Redirects. -->
+     <ItemGroup>
+        <_FileNamesInReferenceCopyLocalPaths Include="@(ReferenceCopyLocalPaths->'%(Filename)%(Extension)')">
+          <SourcePath>%(ReferenceCopyLocalPaths.Identity)</SourcePath>
+        </_FileNamesInReferenceCopyLocalPaths>
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
+          <SourcePath>%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)</SourcePath>
+        </_FileNamesInTestCopyLocalByFileNameWithoutDuplicates>
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" Exclude="@(_FileNamesInTestCopyLocalByFileNameWithoutDuplicates)" />
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" />
+        <_FilteredAssembliesForBindingRedirect Include="%(_FilteredListForBindingRedirect.SourcePath)" Condition="$([System.String]::Copy('%(_FilteredListForBindingRedirect.SourcePath)').EndsWith('.dll')) or $([System.String]::Copy('%(Remaining.SourcePath)').EndsWith('.exe'))" />
```

Comment:
This String.Copy trick is so horrible I opened a suggestion https://github.com/Microsoft/msbuild/issues/1465